src/physics/cam/yog_intr.F90

```diff
@@ -0,0 +1,233 @@
+module yog_intr
+!---------------------------------------------------------------------------------
+! Purpose:
+!
+! CAM interface to the Yoval-O'Gorman deep convection scheme.
+!   based off of the Zhang-McFarlane deep convection scheme
+!
+! Author: J.W. Atkinson
+! January 2024
+!---------------------------------------------------------------------------------
+   use shr_kind_mod, only: r8=>shr_kind_r8
+   use physconst,    only: cpair                              
+   use ppgrid,       only: pver, pcols, pverp, begchunk, endchunk
+   use cam_abortutils,   only: endrun
+   use physconst,        only: pi
+   use spmd_utils,       only: masterproc
+   use perf_mod
+   use cam_logfile,  only: iulog
+   use constituents, only: cnst_add
+   
+   implicit none
+   private
+   save
+
+   ! Public methods
+
+   public ::&
+      yog_readnl,             &! read namelist for YOG scheme/module
+      yog_init,               &! initialize YOG scheme/module
+      yog_final,              &! finalize YOG scheme/module
+      yog_tend                 ! return tendencies
+
+   character(len=136) :: nn_weights    ! location of weights for the YOG NN, set in namelist
```

Comment:
As discussed: should be renamed to `yog_nn_weights`, to avoid naming conflicts in future (if other ML schemes are implemented)

Reply:
Implemented in https://github.com/m2lines/CAM-ML/pull/14/commits/018811553c652da5fe2d9971ddea4cedd66b7f4e